server/container_create_linux.go

```diff
@@ -902,6 +902,9 @@ func (s *Server) createSandboxContainer(ctx context.Context, containerID string,
 		for _, gidmap := range s.defaultIDMappings.GIDs() {
 			specgen.AddLinuxGIDMapping(uint32(gidmap.HostID), uint32(gidmap.ContainerID), uint32(gidmap.Size))
 		}
+		for _, gid := range specgen.Config.Process.User.AdditionalGids {
+			specgen.AddLinuxGIDMapping(gid, gid, uint32(1))
+		}
```

Comment:
should we prevent that these GIDs are added by default in the previous `specgen.AddLinuxGIDMapping`?

Reply:
@giuseppe not sure what you meant by "by default". These are being added only when userns is enabled. May be you meant "should we prevent these GIDs added in the default mappings?" . If this is the case, i am not able to foresee a problem. Can you please elaborate a bit.

Reply:
yes I meant in the "default mappings".

if the `gid` we are adding here was also added as part of (either in the host mapping, or the container mapping):
```
for _, gidmap := range s.defaultIDMappings.GIDs() {
    specgen.AddLinuxGIDMapping(uint32(gidmap.HostID), uint32(gidmap.ContainerID), uint32(gidmap.Size))
}
```

then the runtime will fail to setup the user namespace and the container cannot be created as it is an invalid configuration.

Now we can either try to fix the default mappings to not include any `gid` that is part of `pecgen.Config.Process.User.AdditionalGids` or we could simply fail and give a better error message.  We will need to check that every additional `gid` is not part of:

- [gidmap.HostID; gidmap.HostID+gidmap.Size)
- [gidmap.ContainerID; gidmap.ContainerID+gidmap.Size)

for every `gidmap` in the defaultMappings

Reply:
thanks @giuseppe. will update to have this validation check and error out if invalid